projects/core/src/cart/facade/cart.service.ts

```diff
@@ -223,4 +225,50 @@ export class CartService {
   private isLoggedIn(userId: string): boolean {
     return typeof userId !== 'undefined';
   }
+
+  // TODO: Remove once backend is updated
+  /**
+   * Temporary method to merge guest cart with user cart because of beackend limitation
+   * This is for an edge case
+   */
+  private guestCartMerge(): void {
+    this.getEntries()
+      .pipe(take(1))
+      .subscribe(entries => {
+        this.store.dispatch(
+          new CartActions.DeleteCart({
```

Comment:
I think it's more clear to move this "delete" out of `subscribe`. So, get the entries first, then delete the cart.